tests/test-runner/bin/test-runner.py.in

```diff
@@ -269,6 +270,14 @@ User: %s
         except OSError as e:
             fail('%s' % e)
 
+        """
+        Log each test we run to /dev/kmsg (on Linux), so if there's a kernel
+        warning we'll be able to match it up to a particular test.
+        """
+        if exists("/dev/kmsg"):
+            cmd = "sudo sh -c 'echo ZTS run %s > /dev/kmsg'" % self.pathname
+            os.system(cmd)
```

Comment:
While I don't think this is a bad idea in-of itself, I'm not a fan of this approach since sudo *already* syslogs its argv, so this should end up in the journal all the same already, and this'd double-log (and, if you're on the console, this would render test output absolutely unusable, right?).

Also, you've missed `SUDO` and are using `os.system()`, so either this'd be much better written as
```python
		try:
            kp = Popen([SUDO, "sh", "-c", f"echo ZTS run {self.pathname} > /dev/kmsg"])
            kp.wait()
        except Exception:
            pass
```

Reply:
If I'm not mistaken, the idea here is that by logging to `/dev/kmsg` it's possible to tell exactly what test was executing if the kernel panics.  This can be helpful for the buildbot CI where we're collecting the console logs but not syslogs.  Although I think it would be nice add a flag to make this logging optional.

Reply:
As a non-default flag – yeah, makes sense. In that case my only comment is paragraph 2.

Reply:
Ok, I'll add the Popen() code and make this non-default

Reply:
What is the downside we're trying to avoid by making this non-default? Unless I'm missing something, the reason for making this optional is to prevent people running the tests on the console from getting spammed. Wouldn't that be a relatively rare thing? I'm happy to have this behind a flag, but I think the utility here warrants making it the default.

Reply:
The problem with this is two-fold – (a) it `sudo`s *again* for each test, now logging it *thrice* {3 lines for pam, one for argv} + {3 lines for pam, one for argv} + {1 line of the testname} (and `sudo` isn't exactly free, we have 1500 tests), and (b) yes, it makes the console useless, and I don't think it's unreasonable to expect the simplest way of using a VM to be the one used by a non-negligible amount of people. It makes sense for CI, which is noninteractive and we just get a vague 1000-line dump, but there's no point paying (a) by default (and I don't think the kernel spews warnings by default anyway, so paying for it only when you need it makes more sense to me).

Reply:
I agree, we would enable this in CI after it's in master and the release branches.